win32/devdoc/socket_transport_win32.md

```diff
@@ -0,0 +1,296 @@
+﻿# socket transport Win32
+
+## Overview
+
+socket_transport_win32 is the module that abstracts the socket transport for the windows platform
+
+The `SOCKET_TYPE` determines which API are applicable for this instance of socket_transport.  The following chart shows which API's are allowed under which type:
+
+| API                                    | Allowable Socket Type
+|----------------------------------------|----------------------
+| socket_transport_create                | BOTH
```

Comment:
I kind of have to wonder whether we should enforce this at compile time instead. If we had a `SOCKET_TRANSPORT_CLIENT_HANDLE` and `SOCKET_TRANSPORT_SERVER_HANDLE` and then duplicated the interfaces which say "BOTH" in the table, then you can't call `socket_transport_connect` with `SOCKET_TRANSPORT_SERVER_HANDLE` for example.

Means duplicated function declarations, but the code would be common, and we need the tests for both client and server anyway so it doesn't duplicate too much.

I'm just always in favor of catching errors as close to compile time as possible

Reply:
I had this, at first in the spike, but it starts to get messy because most of the code is common, only 3 functions are different (connect, listen, accept).  It's a lot more code just for 3 function.